vcr/matchers.py

```diff
@@ -30,12 +33,31 @@ def query(r1, r2):
     return r1.query == r2.query
 
 
-def body(r1, r2):
+def raw_body(r1, r2):
     if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
         return r1.body.read() == r2.body.read()
     return r1.body == r2.body
 
 
+def body(r1, r2):
+    if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
+        r1_body = r1.body.read()
```

Comment:
Since you do the exact same thing to r1 and r2, it seems like it might be a good idea to write a function that performs all of these operations.

Think it would help a bunch with readability here.